### PR TITLE
Fix spyglass hyperlink href path

### PR DESCRIPTION
**Pull request by YuviGold**
Related to https://github.com/kubernetes/test-infra/issues/23311. Fixes bug introduced on https://github.com/kubernetes/test-infra/pull/23429 .

Need to remove the quotes, otherwise the path tried to be resolved is
relative to the host while we would like the href link to be absolute.

For example, the bug `href="'${match}'"` caused the link
`http://127.0.0.1:2379`
to redirect to
`<host>/spyglass/static/buildlog/'http://127.0.0.1:2379'`

Removing the quotes fix this issue.

Additionally, fixes the replacement code to change the `innerHTML` and not
the `outerHTML`. This way the `outerHTML` element isn't changed.

### CODE REVIEW

**Review by michelle192837 — APPROVED** [review]
/hold

In case you want Chao's LGTM first, feel free to cancel the hold!

**Review by chaodaiG — APPROVED** [review]
/unhold

**Pull request merged**
